### DISCUSSION TITLE
Screenshot improvements.

### DISCUSSION THREAD

**Discussion by tomkolp** [WeblateOrg/weblate#5655] [Ideas] [answered]
**Describe the solution you'd like**

- Ability to add a screenshot directly from the Projects view or Dashboard. Adding in this way should allow you to search for the source throughout the project. (Useful when we don't know which file contains the text shown in the screenshot.)

- After performing OCR, the captured text should be displayed somewhere, with the possibility of selecting and copying, as there may be situations where the text may not be fully extracted correctly, thus distorting the search results.
(Example 1: The source differs from the screenshot because the program replaces placeholders with other text.
Example 2: The context for each placeholder is in the screenshot, so the source text is different and therefore not found.)
With this feature, it would be possible to copy some text and manually search instead of rewriting.

- It should be possible to include context in search results.
(Example: The context is displayed in the screenshot itself, or we just know about it.)

- Currently, it is difficult for me to determine how accurate OCR in Weblate is at the moment (because I cannot see the content of the captured text). I am not sure if the OCR engine used in Weblate is based on Tesseract, if so, it should be possible to adjust the resolution which will increase the accuracy of the process. Administrators could be allowed to change this value based on their needs and server performance.

- After adding a few screenshots, it would be nice to be able to display the captured strings for translation sorted by the order in which the screenshots were added.

- And when there is a situation of several matching strings for one screen, it would be good if there were links to them somewhere.

**Screenshots**
1
![image](https://user-images.githubusercontent.com/39633242/111071314-de745b00-84d5-11eb-9783-400521b50e97.png)
2
![image](https://user-images.githubusercontent.com/39633242/111071316-e3390f00-84d5-11eb-992d-e4f793989b91.png)
3
![image](https://user-images.githubusercontent.com/39633242/111071329-efbd6780-84d5-11eb-995f-1cc076a4992d.png)
4
![image](https://user-images.githubusercontent.com/39633242/111071335-f64bdf00-84d5-11eb-9873-e5134b04c417.png)
5
![image](https://user-images.githubusercontent.com/39633242/111071341-fc41c000-84d5-11eb-950d-8adad7489271.png)
6
![image](https://user-images.githubusercontent.com/39633242/111071350-0499fb00-84d6-11eb-9b00-80a53c990670.png)

**Comment by nijel** [comment]
Please keep it one topic per issue, having multiple topics in a single issue makes it hard to track. I will convert this to a discussion, please open up separate issues for each feature request if that makes sense.

> Ability to add a screenshot directly from the Projects view or Dashboard.

The screenshots are bound to a component. You can always first search for a string and add a screenshot to it. Doing OCR project or site wide would not scale as it has to fetch all strings to do approximate matching.

> It should be possible to include context in search results.

It can be both searched and is shown in the results:

![obrazek](https://user-images.githubusercontent.com/212189/111073589-9e19da80-84df-11eb-9d8f-568a437fb1cd.png)

> Currently, it is difficult for me to determine how accurate OCR in Weblate is at the moment (because I cannot see the content of the captured text).

The matching process is two step - first is OCR and then similar strings are found.

> I am not sure if the OCR engine used in Weblate is based on Tesseract, if so, it should be possible to adjust the resolution which will increase the accuracy of the process.

Weblate already uses two resolutions to improve the accuracy while doing OCR.

> After adding a few screenshots, it would be nice to be able to display the captured strings for translation sorted by the order in which the screenshots were added.

See https://github.com/WeblateOrg/weblate/issues/3840

> And when there is a situation of several matching strings for one screen, it would be good if there were links to them somewhere.

I don't get this one, sorry.

**Comment by tomkolp** [comment, marked as answer]
Right now, I have about 60,000 screenshots to add. Most of the screenshots match several, and sometimes a dozen or so, components simultaneously. If, when adding one screenshot, I have to use the search engine several times, it does not make sense.
![image](https://user-images.githubusercontent.com/39633242/111075245-572fe300-84e7-11eb-82e9-401acdc92341.png)
The solution from the picture would make it much easier to add one screenshot to several components at the same time.

If this is not possible, I will not create new tickets for each question.

**Reply by nijel** [reply]
Screenshots are currently bound to a component

**Reply by tomkolp** [reply]
OK thank you for your answer. We won't be using screenshots at the moment.
